src/pybind/ceph_volume_client.py

```diff
@@ -1442,6 +1451,34 @@ def get_object(self, pool_name, object_name):
             ioctx.close()
         return bytes_read
 
+    def update_object(self, pool_name, object_name, data_modifier):
+        iterations = 0
+
+        while True:
+            data, version1 = self.get_object_and_version(pool_name, object_name)
+            result_data = data_modifier(data)
+            version2 = self.put_object(pool_name, object_name, result_data)
+            if version2 == version1 + 1:
+                return
+
+            iterations += 1
+            if iterations > 20:
+                raise TooManyVersionMismatch(iterations)
+
+    def get_object_and_version(self, pool_name, object_name):
```

Comment:
Since this is just copy-paste from `def get_object`, I suggest having `get_object` simply call this method and drop the version.

Reply:
Done (if my interpretation is correct).